packages/arb-token-bridge-ui/src/hooks/useAddNewToken.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useState } from 'react'
+import { useAppState } from '../state'
+import { isAddress } from 'ethers/lib/utils'
+
+const useAddNewToken = () => {
+  const [errorMessage, setErrorMessage] = useState('')
+  const [isAddingToken, setIsAddingToken] = useState(false)
+  const {
+    app: {
+      arbTokenBridge: { token }
+    }
+  } = useAppState()
+
+  const storeNewToken = useCallback(
```

Comment:
NIT : If this function is not being used anywhere except in the add flow, why even abstract it out (and useCallback over it) and why not just expand this logic within add function? It is a little bit confusing this hook has 2 member methods - `addNewToken` and `storeNewToken`. 
Pls ignore if we plan to use it elsewhere, but do comment in code :)

Reply:
Make lot of sense 👍, I moved it inside the `try` block